pkg/aggregate/functions.go

```diff
@@ -128,18 +132,19 @@ type LastAggregator struct {
 	lastT int64
 }
 
-func (a *LastAggregator) Clear() { a.val = math.NaN() }
+func (a *LastAggregator) Clear() { a.val = -math.MaxFloat64 } // TODO: use math.Inf(1)
 
 func (a *LastAggregator) Aggregate(t int64, v float64) {
 	if t > a.lastT {
 		a.val = v
 		a.lastT = t
 	}
 }
+
 func (a *LastAggregator) UpdateExpr(col string, bucket int) string {
 	if math.IsNaN(a.val) {
-		// TODO: replace with NaN when engine will support that syntax
-		return fmt.Sprintf("_%s_%s[%d]=%f;", col, a.attr, bucket, -math.MaxFloat64)
+		return ""
```

Comment:
Should remove this `if`. The condition will always evaluate as false.

Reply:
will do it in future PR that will be dealing with replacing NaN by `count` + `Inf` as agreed yesterday. For now I won't remove this safety check because engine does not support NaN yet and there is a scenario when `a.val == NaN`